### PR TITLE
Add AWS PrivateLink Support to Connect Astro to External Data Services

### LINKED ISSUE #985: Document AWS Privatelink as a supported connectivity option
**Request type:**

- [ ] **New document:**: Add the `docs-content` label to this issue.
- [ x] **Addition to an existing document**: Add the `enhancement` label to this issue.

**What is the purpose of the document/enhancement? Select all that apply:**

- [ ] Document a new feature for an upcoming release. Add the appropriate release version label to this issue.
- [ ] Information for an existing feature that isn't yet documented.
- [x ] Additional considerations/ reference material for an existing document.

**Problem/feature that requires documentation:**
We support AWS Privatelink connectivity, alongside VPC Peering and Transit Gateways.  This is not explicitly stated in the docs, so we should add it.  We should also document all the existing Privatelink endpoints that we insert by default into all clusters and mention that we also support some third party endpoints (e.g. Snowflake) 
**Are you the right person to review a draft of the document for technical accuracy? If not, who is?**
Yes. Please also add @astro-walker

### PR DESCRIPTION
**Pull request by kwcanuck**
Resolves #985

### CODE REVIEW

**Review by dnarain-astro — APPROVED** [review]
Looks good to me. Ran it by Jim Weber to make sure he was ok with it, he is as well.

File: astro/connect-external-services.md [hunk 1]
```diff
@@ -92,6 +93,24 @@ If Astronomer creates a new transit gateway in your AWS account for Astro, keep
...
+
+Use AWS PrivateLink to create private connections from Astro to your AWS services without exposing your data to the public internet. 
+
+Astro clusters are pre-configured with the following AWS PrivateLink endpoint services:
+
+- S3 - Gateway Endpoint
+- EC2 Autoscaling - Interface Endpoint
+- ECR - Interface Endpoints for ECR API and Docker Registry API
+- ELB - Interface Endpoint
+- STS - Interface Endpoint
```
Comment (paolaperaza) [inline astro/connect-external-services.md:106]: @kwcanuck For first time references to these services, should we spell out the acronyms? E.g. `Elastic Container Registry (ECR)`?

File: astro/connect-external-services.md [hunk 2]
```diff
@@ -92,6 +93,24 @@ If Astronomer creates a new transit gateway in your AWS account for Astro, keep
...
+
+To request additional endpoints, or assistance connecting to other AWS services, contact [Astronomer support](https://support.astronomer.io).
+
+By default, Astronomer support activates the **Enable DNS Name** option on supported AWS PrivateLink endpoint services.  With this option enabled, you can make requests to the default public DNS service name instead of the public DNS name that is automatically generated by the VPC endpoint service. For example, `*.notebook.us-east-1.sagemaker.aws` instead of `vpce-xxx.notebook.us-east-1.vpce.sagemaker.aws`.
+
+You'll incur an additional AWS infrastructure charge for every AWS PrivateLink endpoint service that you use.  See [AWS PrivateLink pricing](https://aws.amazon.com/privatelink/pricing/).
```
Comment (paolaperaza) [inline astro/connect-external-services.md:112]:
```suggestion
You'll incur an additional AWS infrastructure costs for every AWS PrivateLink endpoint service that you use.  See [AWS PrivateLink pricing](https://aws.amazon.com/privatelink/pricing/). 
```

**Review by paolaperaza — APPROVED** [review]

File: astro/connect-external-services.md (hunk 1 above)
Comment (kwcanuck) [reply astro/connect-external-services.md:106]: @paolaperaza this issue has been addressed

**Pull request merged**